edge-util/src/Microsoft.Azure.Devices.Edge.Storage/InMemoryDbStore.cs

```diff
@@ -13,106 +15,192 @@ namespace Microsoft.Azure.Devices.Edge.Storage
     /// </summary>
     class InMemoryDbStore : IDbStore
     {
-        // Deliberately using list (instead of a synchronied list like Concurrent dictionary) to 
-        // better simulate acutal key-value stores like RocksDb.
-        readonly List<(byte[], byte[])> keyValues;
+        readonly ItemKeyedCollection keyValues;
```

Comment:
Curious - What is the reason to change from List to ItemKeyedCollection?

Reply:
Performance - KeyedCollection provides near O(1) performance for sequential access and indexed access.

Reply:
There was a comment in there saying we wanted a list because it more closely simulated rocks db. Is that not required anymore?

Reply:
I would like that, but there are problems with a list that RocksDb does not have. So at this point I wanted something as close to RocksDB without requiring it to be identical.